File: CS/WindowsApplication1/Custom Ribbon Control/MyRibbonPanelLayoutCalculator.cs
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Ribbon.ViewInfo;
using System.Drawing;

namespace WindowsApplication1.Custom_Ribbon_Control {
    public class MyRibbonPanelComplexLayoutCalculator : RibbonPanelComplexLayoutCalculator {
        public MyRibbonPanelComplexLayoutCalculator(RibbonPanelViewInfo panelInfo)
            : base(panelInfo) {
        }
        public override void UpdatePanelLayout() {
            int xPos = ContentBounds.Left - PanelInfo.PanelScrollOffset;

            for (int i = 0; i < PanelInfo.Groups.Count; i++) {
                RibbonPageGroup pageGroup = PanelInfo.Groups[i].PageGroup;

                PanelInfo.Groups[i].CalcViewInfo(new Rectangle(xPos, ContentBounds.Top, PanelInfo.Groups[i].PrecalculatedWidth, ContentBounds.Height));

                if (object.Equals(pageGroup.Tag, "AlignRight"))
                    continue;

                xPos += PanelInfo.Groups[i].PrecalculatedWidth + PanelInfo.DefaultIndentBetweenGroups;
            }
        }
    }
}

File: VB/WindowsApplication1/Custom Ribbon Control/MyRibbonPanelLayoutCalculator.vb
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars.Ribbon.ViewInfo
Imports System.Drawing

Namespace WindowsApplication1.Custom_Ribbon_Control
    Public Class MyRibbonPanelComplexLayoutCalculator
        Inherits RibbonPanelComplexLayoutCalculator

        Public Sub New(ByVal panelInfo As RibbonPanelViewInfo)
            MyBase.New(panelInfo)
        End Sub
        Public Overrides Sub UpdatePanelLayout()
            Dim xPos As Integer = ContentBounds.Left - PanelInfo.PanelScrollOffset

            For i As Integer = 0 To PanelInfo.Groups.Count - 1
                Dim pageGroup As RibbonPageGroup = PanelInfo.Groups(i).PageGroup

                PanelInfo.Groups(i).CalcViewInfo(New Rectangle(xPos, ContentBounds.Top, PanelInfo.Groups(i).PrecalculatedWidth, ContentBounds.Height))

                If Object.Equals(pageGroup.Tag, "AlignRight") Then
                    Continue For
                End If

                xPos += PanelInfo.Groups(i).PrecalculatedWidth + PanelInfo.DefaultIndentBetweenGroups
            Next i
        End Sub
    End Class
End Namespace
